### PR TITLE
Refactor `AvatarDropdown` Component

### LINKED ISSUE #66: Cleanup and Refactor `AvatarDropdown` component
- Fix styles _(currently the dropshadow is broken, size is inconsistent with Dapp)_
- Update styles with the ones from the Dapp (including the responsive styles)
- Remove the need for colony data to be passed into it
- re-enable all the pertinent menu entries

This should be done either together with #63 or wait for it to be done

### PR DESCRIPTION
**Pull request by Julianahlk**
## Description

This PR focuses on the refactoring and cleaning up the AvatarDropdown component and other relating components.  

**Changes** 🏗

* Ported `AvatarDropdown` styles for mobile
* Ported `PopOverSection`
* Ported `AvatarDropdownPopoverMobile`
* Added `ItemContainer`

## TODO

- [x] Fix styles (currently the drop shadow is broken, size is inconsistent with Dapp)
- [x] Update styles that match the ones from the Dapp (including the responsive styles)
- [x] Remove the need for colony data to be passed into it
- [x] Re-enable all the pertinent menu entries (for mobile this would be done in the HamburgerDropdown)
- [x] Add a task to port over the [HamburgerDropdown](https://github.com/JoinColony/colonyCDapp/issues/107) menu for mobile
- [x] Add a task to update the [AvatarDropdownPopoverMobile](https://github.com/JoinColony/colonyCDapp/issues/108) once tokens are ready
 
Resolves #66

### CODE REVIEW

File: src/components/frame/RouteLayouts/UserNavigation/UserNavigation.tsx [hunk 1]
```diff
@@ -23,16 +25,26 @@ const MSG = defineMessages({
...
 });
 
+const { query700: query } = queries;
+
 const UserNavigation = () => {
   const { colony } = useColonyContext();
-
   const { wallet } = useAppContext();
   const { formatMessage } = useIntl();
+  const isMobile = useMediaQuery({ query });
```
Comment (jakubcolony) [inline src/components/frame/RouteLayouts/UserNavigation/UserNavigation.tsx:43]: It seems like we're following the same pattern in multiple files:
- Import `queries`
- Destructure the `query700`
- Pass it to `useMediaQuery` to get `isMobile`

Maybe it's worth refactoring it into its own hook?

File: src/components/shared/PopoverSection/UserSection.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          </DropdownMenuItem>
+        </>
+      ) : (
+        <DropdownMenuItem>
+          <NavLink
+            to={{
+              pathname: CREATE_USER_ROUTE,
+              state: colony?.name
+                ? { colonyURL: `/colony/${colony?.name}` }
+                : {},
```
Comment (jakubcolony) [inline src/components/shared/PopoverSection/UserSection.tsx:65]: Is this still needed? Looking at (Will's User Creation PR)[https://github.com/JoinColony/colonyCDapp/pull/88/], he doesn't seem to use it anywhere.
In the Dapp it was used in `WalletRequiredRoute` which we've abandoned in the CDapp

**Review by jakubcolony — APPROVED** [review]
Looking great, thumbs up from me! 💪 
image
image

File: src/components/shared/PopoverSection/UserSection.tsx (hunk 2 above)
Comment (Julianahlk) [reply src/components/shared/PopoverSection/UserSection.tsx]: I think you are right, updated 👍

File: src/components/frame/RouteLayouts/UserNavigation/UserNavigation.tsx (hunk 1 above)
Comment (Julianahlk) [reply src/components/frame/RouteLayouts/UserNavigation/UserNavigation.tsx]: Sounds good, added a hook for this `useMobile`

File: src/hooks/useMobile.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { useMediaQuery } from 'react-responsive';
+
+import queries from '~styles/queries.css';
+
+const useMobile = () => {
```
Comment (chinins) [inline src/hooks/useMobile.ts:5]: great idea!

**Review by chinins — APPROVED** [review]
Great job!

**Pull request merged**